### DISCUSSION TITLE
General Help Connecting Cameras

### DISCUSSION THREAD

**Discussion by lueghamero** [ImSwitch/ImSwitch#213] [Q&A]
Hello Everyone, 

I'm currently trying to figure out ImSwitch by connecting 3 cameras to it, which I later want to use for focus locking and imaging. 
However I am having a hard time to get it done as I run into problems with every single one of them. 

For example I want to use a Daheng camera for the focus lock. There is already code for that camera in the dev branch for example. I moved the gxipy folder and gxipycamera.py to the interfaces folder, as well as the GXIPIPYManager to detectors. Everything should be set up correctly including the json file. However I always receive the error that gxipy module could not be found. 
I already tried adding it to the path and also importing it in test files such as the GxSingleCamMono (which is also located in interfaces folder) is working fine.
Can anyone help me? I am out of Ideas, I don't understand why it would not load the gxipy when running the main.py file. 

Cheers

**Comment by beniroquai** [comment]
Hey @lueghamero, thanks for reaching out. I have implemented the Daheng Stack. It's working in the https://github.com/openUC2/ImSwitch/tree/master branch just fine. This got merged into the imswitch/imswitch dev branch recently. The only Issues I can think if is:
1. wrong camera ID 
2. the path to the DLLs is wrong. 

1.- > Same as in the example python code you used from the Daheng SDK
2. -> How did you install imswitch? You have to do `git clone XXXX` in order to get the DLLs: https://github.com/openUC2/ImSwitch/tree/master/imswitch/imcontrol/model/interfaces/gxipy/dll, set a breakpoint here: https://github.com/openUC2/ImSwitch/blob/master/imswitch/imcontrol/model/interfaces/gxipycamera.py#L8 and see what's happening down the road. I'm 99% sure it's a missing DLL error which would be loaded successfully here https://github.com/openUC2/ImSwitch/blob/master/imswitch/imcontrol/model/interfaces/gxipy/gxwrapper.py#L16

**Reply by lueghamero** [reply]
Hi @beniroquai thank you for your help. I have tried out a lot of stuff already, from giving the path explicitly to checking baud rate and camera ID. The only thing for the Daheng camera that worked for me was downloading your repository and running the daheng_histogram setup. 
The thing is, as soon as i want to run any other config, for example the  coolLED or the sted, I will always get Key errors. Also when trying to implement a manager for example for a Photon Focus camera which i want to control via pylablib I will always end up with a Key error. I use the correct python version in my environment and have all the packages. 
If I use the exact same environment with the repository from https://github.com/ImSwitch/ImSwitch and run my Manager for the Photon Focus it will connect to the camera. 
I really don't see what is causing these problems.

**Reply by beniroquai** [reply]
What exactly do you mean by keyerror? Can you pass the stacktrace please?
